### ISSUE TITLE
Vertex color editor for Polygon2D

### ISSUE THREAD

**Issue by aXu-AP** [godotengine/godot-proposals#8195]
### Describe the project you are working on

Looking to resurrect an old 2D platformer game which uses textured polygons for characters and environments.

### Describe the problem or limitation you are having in your project

Polygon2D supports vertex colors but there's no way to edit this data outside of setting it manually or by code. Vertex colors are useful for coloring polygons (duh), and for various vfx like adjusting amount of wind affecing leaves.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Inside UV editor, there would be a Color mode. User would be able to apply color to their vertices with brush and fill tools.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Color editor would support following functionality:
![kuva](https://github.com/godotengine/godot-proposals/assets/1621768/deec5004-3cf0-49b2-9e39-57b8c15b2d1d)
* Adjust all channels at the same time (good for coloring the polygon) or one channel at the time (good for using shaders that interpret the colors as something else).
* Brush tool. Paints vertices' color to selected value within radius of mouse cursor. Allow to adjust size and hardness.
* Bucket tool. Paint all connected polygons at once.
* Opacity. Set to less to make minor adjustments instead of overwriting current colors. Note: Setting color to transparent sets the vertices themselves to be transparent, so this needs to be a separate option.
Small note: on the above mockup I use sliders but in the actual editor I would use EditorSpinSlider for compactness.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Used more or less as often as uv and weight editor.

### Is there a reason why this should be core and not an add-on in the asset library?

Adds editor support for existing core functionality.

**Comment by Calinou** [comment]
We already have a Polygon2D UV editor. I wonder if both should be merged in the same dialog (and rename the tool to "Polygon2D attribute editor" in the process).

**Comment by aXu-AP** [comment]
Yes, this is exactly my intention, see the part "Inside UV editor, there would be a Color mode."

Altough it's called UV editor, it's also bone weight and and general polygon editor, so renaming it would make sense.

**Comment by aXu-AP** [comment]
Made a separate proposal for bottom dock & refactoring to reflect editor's intention better: #8209 
Refactor would help adding Color editor on the road, as currently magic numbers are used for distincting between modes, suggested change to use enum.
